lib/Role/utils.js

```diff
@@ -0,0 +1,167 @@
+export function getUpdatedDisabledCapabilities(newDisabledList, initialDisabledCapabilities, isChecked) {
+  const result = { ...initialDisabledCapabilities };
+  const add = (item, currentCount) => {
+    result[item] = currentCount + 1;
+  };
+  const remove = (item, currentCount) => {
+    if (currentCount > 1) {
+      result[item] = currentCount - 1;
+    } else {
+      delete result[item];
+    }
+  };
+
+  newDisabledList.forEach(item => {
+    const currentCount = result[item] || 0;
+    if (isChecked) {
+      add(item, currentCount);
+    } else {
+      remove(item, currentCount);
+    }
+  });
+
+  return result;
+}
+
+export function batchUpdatedDisabledCapabilities({
+  prevDisabledCapabilities,
+  selectedCapabilitySets,
+  capabilitySetsList,
+  checked,
+}) {
+  let result = { ...prevDisabledCapabilities };
+  selectedCapabilitySets.forEach((cap) => {
+    const found = capabilitySetsList.find((c) => c.id === cap);
+    if (!found) return;
+
+    result = getUpdatedDisabledCapabilities(
+      found.capabilities,
+      result,
+      checked,
+    );
+  });
+
+  return result;
+}
+
+export function useCheckboxHandlers({
```

Comment:
pedantic: be careful with the `use...` prefix since it it's how hooks are named, which could give somebody the wrong idea about what this function does. As soon as you look, it clearly isn't a hook, but it's sort of like having a cat named "Dog". Maybe `getCheckboxHandlers` or `checkboxHandlersFor()`.